### ISSUE TITLE
[Bugfix] Investigate bytes lost maybe in IsoTP

### ISSUE THREAD

**Issue by Nahrain1** [amosproj/amos2024ss07-updating-flash-boot-loader#75]
## User story
1. As a Software Developer
2. I want to investigate and fix the issue of bytes being lost during transmission
3. So that messages are transmitted completely and accurately

## Acceptance criteria
* The conditions under which the last byte of a message gets lost are identified
* IsoTP layer is checked to ensure if it is the origin 
* A fix to ensure that all bytes, including the last byte, are correctly transmitted is implemented
* The fix is tested with various message lengths and conditions 
* The fix does not impact the performance or reliability of the IsoTP layer

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

**Comment by sebas-rodri** [comment]
@wlklsn and I removed padding of CAN messages on MCU side, it seems like the GUI cant handle padded CAN Messages.
